Automattic/wp-calypso
# Add Jetpack Cloud Server Connection Indicator Component

#### Changes proposed in this Pull Request

* Add `getRewindStateRequestStatus` selector
* Add new `ServerConnectionIndicator` Jetpack Cloud Component
* Add the component  to Jetpack Cloud `/settings/:site` route and refactor page in functional style

**Narrow ( not quite mobile ) screenshots**
<img width="400" alt="narrow connected screenshot" src="https://user-images.githubusercontent.com/2810519/75709987-94c52380-5c78-11ea-98f2-af7376d0c95d.png">
<img width="400" alt="narrow not-connected screenshot" src="https://user-images.githubusercontent.com/2810519/75709969-91319c80-5c78-11ea-99f4-0777447e78f7.png">
**Wide Screenshots**
<img width="400" alt="wide connected screenshot" src="https://user-images.githubusercontent.com/2810519/75710038-a9092080-5c78-11ea-8823-e8a309c34aaa.png">
<img width="400" alt="wide not-connected screenshot" src="https://user-images.githubusercontent.com/2810519/75710047-ab6b7a80-5c78-11ea-8989-d873b29c00e8.png">

#### Testing instructions



1. Navigate to `/settings/:site` for a Jetpack site with some form of backup
2. Verify that after loading the page matches on of the above screenshots
3. [Using Redux DevTools dispatch the following action](https://github.com/reduxjs/redux-devtools)
```js
{
    type: 'REWIND_STATE_UPDATE',
    siteId: [your site id],
    data: {
        state: 'active'
    }
}
```
where `[your site id]` is your site id
4. Verify that the page matched the "Server Status: Connected" screenshot above
5. Using Redux DevTools dispatch the following action
```js
{
    type: 'REWIND_STATE_UPDATE',
    siteId: [your site id],
    data: {
        state: 'not-active'
    }
}
```
6. Verify that the page matched the "Server Status: Not connected" screenshot above
7. Verify that the Daily or Real-time on the page matches the plan for the site
8. Repeat steps 5-7 with a Jetpack site with the opposite type of plan
